verifier/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+mod checks;
+mod commitment;
+mod error;
+mod label_encoder;
+mod pubkey;
+mod signed;
+mod tls_doc;
+mod utils;
+mod verifier_doc;
+mod webpki_utils;
+
+use crate::signed::Signed;
+use error::Error;
+use pubkey::PubKey;
+use verifier_doc::{VerifierDoc, VerifierDocUnchecked};
+
+type HashCommitment = [u8; 32];
+
+/// A PRG seeds from which to generate garbled circuit active labels, see
+/// [crate::commitment::CommitmentType::labels_blake3]
+type LabelSeed = [u8; 32];
+/// Verifier of the notarization document
+///
+/// Once the verification succeeds, an application level (e.g. HTTP, JSON) parser can
+/// parse `commitment_openings` in `doc`
+struct Verifier {
+    /// A validated notarization document which needs to be verified
+    doc: VerifierDoc,
+    /// A trusted Notary's pubkey (if this Verifier acted as the Notary then no pubkey needs
+    /// to be provided)
+    trusted_pubkey: Option<PubKey>,
+}
```

Comment:
I have no opinion on this matter because I don't know exactly how the VerifierDoc will be used. Maybe it makes sense to keep it inside the `Verifier`, so that http parser would access the doc via verifier.doc() (we will have to make sure that this method returns the doc only after .verify() suceeds).
Otherwise, there is a risk of the parser accidentally taking an unverified doc and working with it.

Reply:
Or maybe even better, we could have a VerifiedDoc type ?

Reply:
I introduces the VerifiedDoc type but I do not agree that there should be no `Verifier` type, since not having it would hurt readability.